### ISSUE TITLE
Figure numbering across the map

### ISSUE THREAD

**Issue by jelovirt** [dita-ot/dita-ot#1129]
Converted from [SourceForge issue 1903166](http://sourceforge.net/support/tracker.php?aid=1903166), submitted by joezucker

For HTML output, currently the numbering scheme for figures and tables restart for each generated HTML file.  

This request is to support the abiltiy to designate consecutive numbering across topics (specifically HTML numbering crosses the entire map).  This should be done as an option so that users do not get it by default.

For addtional information about this request, please see #290

**Comment by nakohdo** [comment]
I'm encountering the same behavior with PDF output (DITA-OT 1.8.5). Any news on this issues? And how does oXygen get around this problem? The PDF transformation run from oXygen numbers figures for the whole document.

**Comment by drmacro** [comment]
For PDF there's no reason for not getting numbering across the publication since you have the merged doc available.

For HTML, you need something like the D4P map-driven processing framework, which is used by the D4P HTML2, HTML5, and EPUB transforms. It processes the map to construct an "enumeration" data file, which reflects the publication structure with just enough detail to allow numbering without literally merging the topics as is done for the PDF. All topic processing is done in the context of the map so that you know the use of context of the topic and can therefore locate it within the enumeration structure and number things appropriately.

The OT explicitly avoids this by only doing map-based processing during preprocessing.

In theory you could adapt the D4P enumeration processing to preprocessing but you wouldn't have access to the map during topic processing, which is also required.

Basically, the OTB OT processing is optimized to minimize memory usage at the cost of flexibility. The D4P processing is optimized for flexibility and publication-aware processing at the cost of increased memory usage.

**Comment by nakohdo** [comment]
Thanks for the quick reply, Eliot. We've been digging a bit deeper into the issue and it looks like a glitch in our customization. I will keep you posted.
